### ISSUE TITLE
Is there a limit on the number of entries?

### ISSUE THREAD

**Issue by lasergnu** [Zipios/Zipios#25]
I found that this does not work:
touch {1..65536} && zip rr.zip * && zipios_example rr.zip 1

But this does:
touch {1..65535} && zip rr.zip * && zipios_example rr.zip 1

Which hints that the issues arise because zipios does not support ZIP64

**Comment by AlexisWilke** [comment]
Hi lasergnu, I suppose this is in link with issue #24 correct? You should have commented on the other issue instead of adding a new one. And yes, there may  be something that changes whenever we reach a certain number of files and zipios may not yet be aware of that variation in a .zip file. Can you confirm that the file you test with in #24 is also one that has many entries? (64K sounds about right as a break point.)

**Comment by dooglio** [comment]
Patches welcome. :-)

**Comment by lasergnu** [comment]
No, both files I test in #24 only have a small handful of entries.  Hence why I assumed this was two separate issues.  

You can reproduce the issue with the example Java code in the link.  I did some reading and found that JDK 8 uses ZIP64 by default so it could be the exact same issue after all...

**Comment by AlexisWilke** [comment]
Ah, yes. A Zip64 header would do it. It's not compatible with the older format. It makes sense to create a Zip64 header by default so as to support any size. It's much harder to create a 32 bit and once you reach the limit switch to 64 because you then need to edit the entire file to use 64 bit structures.
